### ISSUE TITLE
When Saving, Prompt To Kill Emacs

### ISSUE THREAD

**Issue by dmizelle** [hlissner/doom-emacs#4903]
**What did you expect to happen?**
When editing certain files (so far I've noticed Golang and YAML), I expect hitting `Z-Z` or `:wq` to save the file and exit cleanly

**What actually happened?**
Sometimes when saving specific file formats (in this case Golang and YAML) I end up seeing a message at the bottom of the screen asking:

```
Buffer kustomization.yaml modified; kill anyway? (y or n)
```

I have to hit `n` or else the contents of my changes are lost.

**Additional details:**
I'm running emacs in server/client mode. This strangely doesn't happen in just invoking emacs on the command line via `emacs -nw`

```
❯ cat /usr/lib/systemd/user/emacs.service
[Unit]
Description=Emacs text editor
Documentation=info:emacs man:emacs(1) https://gnu.org/software/emacs/

[Service]
Type=notify
ExecStart=/usr/bin/emacs --fg-daemon
ExecStop=/usr/bin/emacsclient --eval "(kill-emacs)"
# The location of the SSH auth socket varies by distribution, and some
# set it from PAM, so don't override by default.
# Environment=SSH_AUTH_SOCK=%t/keyring/ssh
Restart=on-failure

[Install]
WantedBy=default.target
```

I then call `emacsclient` through an alias, because I'm a curmudgeon with old habits that die hard.

```
❯ which vim
vim: aliased to emacsclient -nw -c
```

Configs: https://gist.github.com/dmizelle/7aca86e86186dfbb9ae4f33e9a5d6fb8

Recording: https://asciinema.org/a/CTaXfoGszHbkDB8CQMjpoMWdk

Key sequences:
* After editing a file, a simple `ZZ` or `:wq` is performed.

```
Ignoring ’:ensure t’ in ’doom-modeline’ config
Ignoring ’:ensure t’ in ’rego-mode’ config
Starting Emacs daemon.
Doom loaded 311 packages across 65 modules in 2.287s
Package cl is deprecated
Loading /home/devon/.emacs.d/.local/cache/recentf...done
(New file)
Undo-Fu-Session discarding undo data: file length mismatch for "/tmp/test.yaml"
When done with a buffer, type C-x #
Company-Ispell: lookup-words error: No plain word-list found at systemdefault locations.  Customize ‘ispell-alternate-dictionary’ to set yours.
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Buffer test.yaml modified; kill anyway? (y or n) n
 [1] main  | Deleted '#1' workspace
Reformatted with prettier
Wrote /tmp/test.yaml
"test.yaml" 7L, 57C written
(New file)
When done with a buffer, type C-x #
 [1] main  | Deleted '#1' workspace [2 times]
No server buffers remain to edit
Note: standard-indent, tab-width, evil-shift-width adjusted to 4
When done with a buffer, type C-x #
Quit
Buffer test.yaml modified; kill anyway? (y or n) y
 [1] main  | Deleted '#1' workspace
Already formatted
QuitInvalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Open documentation (doom/help)
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
Invalid face attribute :background nil
Invalid face reference: hl-line
```

**Steps to reproduce:**
1. Start `emacs` in the background with `emacs --fg-daemon`
2. Edit a YAML (also happens with Golang so far) file with `emacsclient -nw -c $file`
3. Add another line
4. Press `ZZ` or `:wq`
5. Observe the message at the bottom of the frame

**System information:**
https://gist.github.com/dmizelle/f9dd2b690bdd79f3330aa8e98651a48c

(antergos is just a hold over from the project, its was Arch Linux with a simple installer)

**Comment by hlissner** [comment]
I cannot reproduce this in vanilla Doom, on 26.3, 27.1, 27.2 or 28 (with or without native-comp), on Arch Linux, NixOS, or macOS, so it must be an issue with your config.

I will close this as there's no reason to suspect this is a Doom issue. I will reopen it if I find a new lead that suggests otherwise.

**Comment by dmizelle** [comment]
Okay! Thank you very much for taking a look, and for doom. :)

If I find out what it was, I'll comment here in case some errant soul's Google search takes them to this issue.

**Comment by dmizelle** [comment]
I ended up figuring out that this only happened if I had `+onsave` set for `(format)` in `init.el`.

If I disabled that option, I no longer had this behavior when `ZZ`-ing. However, if I hit `SPC c-f`, then it would happen again.

For giggles, and to try something, I switched to gccemacs (specifically the `emacs-native-comp-git-enhanced` package in the AUR if any other Arch users come by). After running a `doom sync`, the behavior no longer happens.

I have zero clue why it would make any difference, but I'm happy that it no longer happens.

Thanks for the peek again, and thanks for your work.
